### ISSUE TITLE
Use v8 codebase instead of v6

### ISSUE THREAD

**Issue by talentlessguy** [dhruvkelawala/web3-starknet-react#5]
Hey, 

I've gone through the code and noticed that it has a similar structure to web3-react v6, which has lots of issues and bugs, and is less performant.

What do you think of using v8 codebase instead?

Pros for using it:
- it's actively maintained
- less junk there (smaller size)
- more performant and granular thanks to Zustand

**Comment by dhruvkelawala** [comment]
> Hey, 
> 
> 
> 
> I've gone through the code and noticed that it has a similar structure to web3-react v6, which has lots of issues and bugs, and is less performant.
> 
> 
> 
> What do you think of using v8 codebase instead?
> 
> 
> 
> Pros for using it:
> 
> - it's actively maintained
> 
> - less junk there (smaller size)
> 
> - more performant and granular thanks to Zustand

Yes, you are right, it's based on v6. I am not sure what bugs and performance issues this version has but when I started working on this repo, v6 was the default branch as v8 was still in works. v8 is still currently in beta and projects like Uniswap are still using v6 afaik.

Having said that, I will create a new branch on this repo ( let's say v2 ) and we can use web3-react's v8 structure and see how it works. 

Thanks for the suggestions. Highly appreciated. 🙂
